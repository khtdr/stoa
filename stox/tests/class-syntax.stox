#!/usr/bin/env stox

class Breakfast {
  cook() {
    print "Fried eggs.";
  }
  serve(who) {
    print "Enjoy, " + who + ".";
  }
}

class Bagel {}
var bagel = Bagel();

bagel.isReady = false;
print bagel.isReady ? "Bagel is ready" : "... still toasting...";
bagel.isReady = true;
print bagel.isReady ? "Bagel is ready" : "... still toasting...";

// Breakfast().cook().serve('Customer');
