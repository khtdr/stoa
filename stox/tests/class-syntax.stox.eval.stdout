... still toasting...
Bagel is ready
